src/sentry/stacktraces/processing.py

```diff
@@ -162,26 +168,36 @@ def preprocess_step(self, processing_task):
         return False
 
 
-def find_stacktraces_in_data(data, include_raw=False):
+def find_stacktraces_in_data(data, include_raw=False, with_exceptions=False):
     """Finds all stracktraces in a given data blob and returns it
     together with some meta information.
 
-    If `include_raw` is True, then also raw stacktraces are included.
+    If `include_raw` is True, then also raw stacktraces are included.  If
+    `with_exceptions` is set to `True` then stacktraces of the exception
+    are always included and the `is_exception` flag is set on that stack
+    info object.
     """
     rv = []
 
-    def _report_stack(stacktrace, container):
-        if not stacktrace or not get_path(stacktrace, "frames", filter=True):
+    def _report_stack(stacktrace, container, is_exception=False):
+        if not is_exception and (not stacktrace or not get_path(stacktrace, "frames", filter=True)):
```

Comment:
This change is indeed confusing as we're now reporting non-existing stack traces just to get to the exception container. I also don't have a better idea, since I suppose you need the stack frame's platform to decide whether you want to touch the container exception. 

To be safe, you could default `frames` to an empty list, but since you only selectively turn this on, I could not find an invalidated invariant. At least the native plugin relies on empty stacktraces being removed, but there you don't toggle, so it's fine. 

As an alternative, could we add a `find_exceptions_in_data` and call it in `should_process_for_stacktraces` as well as `process_stacktraces` before processing the stack traces?

Reply:
I would like to change the entire thing to return stacks + related data and treat missing stacks as empty stacks.